### DISCUSSION TITLE
Ability to search on filename or content on the one page

### DISCUSSION THREAD

**Discussion by clayjam** [microsoft-search/pnp-modern-search#1955] [Q&A] [answered]
Providing users with the ability to search only on the filename provides many benefits. I am able to complete this using the v4 web parts using Managed Properties. What I want is a way for users to search filenames via one search box and content via another search box on the same page. This would provide great flexibility to users. If this isn't achievable can you think of another way I can achieve this ?

One option is to use Search Verticals web part and have one page configured for filename and the other vertical configured for all content but I would prefer this to be on one page.

**Comment by wobba** [comment, marked as answer]
Using the vertical setup might be better where you have a different query template per web parts using `Filename:({searchTerms})` for the filename one.

**Reply by clayjam** [reply]
Thanks - this is what I thought. Would be great if we could have two search box web parts on the same page to manage this.
